drivers/dai/nxp/micfil/micfil.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/drivers/dai.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include "fsl_pdm.h"
+
+#define DT_DRV_COMPAT nxp_dai_micfil
+LOG_MODULE_REGISTER(nxp_dai_micfil);
+
+#define MICFIL_CLK_ROOT 24576000
+#define MICFIL_OSR_DEFAULT	16
+
+#define UINT_TO_MICFIL(x) ((PDM_Type *)(uintptr_t)(x))
+
+#define MICFIL_FIFO_BASE(inst) \
+	POINTER_TO_UINT(&(UINT_TO_MICFIL(DT_INST_REG_ADDR(inst))->DATACH[0]))
+
+#define MICFIL_DMA_HANDSHAKE(inst, dir) \
+	((DT_INST_DMAS_CELL_BY_NAME(inst, dir, channel) & GENMASK(7, 0)) | \
+	((DT_INST_DMAS_CELL_BY_NAME(inst, dir, mux) << 8) & GENMASK(15, 8)))
+
+#define MICFIL_FIFO_DEPTH(inst) DT_INST_PROP(inst, fifo_depth)
+
+struct dai_nxp_micfil_data {
+	PDM_Type *base;
+	struct dai_config cfg;
+};
+
+struct dai_nxp_micfil_config {
+	const struct dai_properties *rx_props;
+	uint32_t fifo_depth;
+};
+
+/* this needs to match SOF struct sof_ipc_dai_micfil_params */
+struct micfil_bespoke_config {
+	uint32_t pdm_rate;
+	uint32_t pdm_ch;
+};
+
+static void dai_nxp_micfil_trigger_start(const struct device *dev)
+{
+	struct dai_nxp_micfil_data *micfil_data = dev->data;
+
+	PDM_Reset(micfil_data->base);
```

Comment:
is it a good idea to restart the module on TRIG_START? what if we PAUSE-RESUME? won't this lead to data loss?

Reply:
Yes, I think you are right. This pattern (of resetting the device at start) is followed by the Linux kernel also but they save the registers at suspend and restore them at resume.

So, I think I will reset the device at init() once only. Also, we use the software reset happens in PDM_Init() which is called by set_config() so I think it is not even necessary in start().

Reply:
pls drop if not needed. Otherwise, I'd suggest doing it in `config()` unless there's some other reason we'd need to ungate clocks in init().